fusillade-api.yml

```diff
@@ -4,164 +4,465 @@ info:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                policy:
+                  type: string
+      responses:
+        200:
+          description: policy modified
+        401:
+          description: The group is not authorized because the object has already been created and the user does not have permission to modify, or user does not have permission to create objects.
         default:
           description: Unexpected error
   /groups/{group_id}:
     get:
-      summary: Retrieve a group.
+      summary: Get properties of a group
       security:
-        - openId:
-            - https://www.googleapis.com/auth/userinfo.email
+        - bearerAuth: []
       parameters:
-        - name: group_id
-          in: path
-          required: true
-          type: string
+        - $ref: '#/components/parameters/group_id'
       responses:
         200:
-          description: OK
-          schema:
-            type: object
+          description: Information about the role is returned
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Group'
+        404:
+          description: Resource not found.
         default:
           description: Unexpected error
-    delete:
-      summary: Delete a group.
+    put:
+      summary: Create a new group
       security:
-        - openId:
-            - https://www.googleapis.com/auth/userinfo.email
+        - bearerAuth: []
       parameters:
-        - name: group_id
-          in: path
-          required: true
-          type: string
+        - $ref: '#/components/parameters/group_id'
+        - $ref: '#/components/parameters/action'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                policy:
+                  type: string
+      responses:
+        200:
+          description: Group was created
+        401:
+          description: The user is not authorized because the object has already been created and the user does not have permission to modify, or user does not have permission to create objects.
+        default:
+          description: Unexpected error
+  /groups/{group_id}/roles:
+    get:
+      summary: Retrieve roles a group is in.
+      security:
+        - bearerAuth: []
+      parameters:
+        - $ref: '#/components/parameters/group_id'
       responses:
         200:
           description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  - role:
+                      type: array
+                      items:
+                        type: string
+        404:
+          description: Resource not found.
+        default:
+          description: Unexpected error
+    put:
+      summary: Add group to roles
```

Comment:
I'm unfamiliar with fusillade... are you allowing a group to assume a role, like AWS or is there some larger "roles" machinery that groups are added to?

Reply:
changed to add roles to group